bigquery/snippets/managedwriter/bigquerystorage_append_rows_pending.go

```diff
@@ -0,0 +1,205 @@
+
+		// This is a required field, and thus must be present.
+		RowNum: proto.Int64(23),
+
+		// StructCol is a single nested message.
+		StructCol: &exampleproto.SampleData_SampleStruct{
+			SubIntCol: proto.Int64(random.Int63()),
+		},
+
+		// StructList is a repeated array of a nested message.
+		StructList: sList,
+	}
+
+	// Now that the protocol message has been populated, serialize it to binary form and return.
+	return proto.Marshal(m)
+}
+
+// appendToPendingStream demonstrates using the managedwriter package to write some example data
+// to a pending stream, and then committing it to a table.
+func appendToPendingStream(w io.Writer, projectID, datasetID, tableID string) error {
+	// projectID := "myproject"
+	// datasetID := "mydataset"
+	// tableID := "mytable"
+
+	ctx := context.Background()
+	client, err := managedwriter.NewClient(ctx, projectID)
+	if err != nil {
+		return fmt.Errorf("managedwriter.NewClient: %v", err)
+	}
+
+	// Create a new pending stream.  We'll use the stream name to construct a writer.
+	pendingStream, err := client.CreateWriteStream(ctx, &storagepb.CreateWriteStreamRequest{
+		Parent: fmt.Sprintf("projects/%s/datasets/%s/tables/%s", projectID, datasetID, tableID),
+		WriteStream: &storagepb.WriteStream{
+			Type: storagepb.WriteStream_PENDING,
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("CreateWriteStream: %v", err)
+	}
+
+	// We need to communicate the descriptor of the protocol buffer message we're using, which
+	// is analagous to the "schema" for the message.  This example uses a self-contained message, but
+	// we'll call adapt.NormalizeDescriptor which will correct any descriptors that aren't
+	// self-contained.
+	m := &exampleproto.SampleData{}
+	descriptorProto, err := adapt.NormalizeDescriptor(m.ProtoReflect().Descriptor())
+	if err != nil {
+		return fmt.Errorf("NormalizeDescriptor: %v", err)
+	}
+
+	// Instantiate a ManagedStream, which manages low level details like connection state and provides
+	// additional features like a future-like callback for appends, etc.  NewManagedStream can also create
+	// the stream on your behalf, but in this example we're being explicit about stream creation.
+	managedStream, err := client.NewManagedStream(ctx, managedwriter.WithStreamName(pendingStream.GetName()),
+		managedwriter.WithSchemaDescriptor(descriptorProto))
+	if err != nil {
+		return fmt.Errorf("NewManagedStream: %v", err)
+	}
+
+	// First, we'll append a single row.
+	rowBytes, err := generateExampleMessage()
+	if err != nil {
+		return fmt.Errorf("generateExampleMessage: %v", err)
+	}
+
+	// we'll keep track of the current offset in the stream with curOffset.
+	var curOffset int64
+
+	result, err := managedStream.AppendRows(ctx, [][]byte{rowBytes}, managedwriter.WithOffset(0))
+	if err != nil {
+		return fmt.Errorf("AppendRows first call error: %v", err)
+	}
+
+	// GetResult blocks until we receive a response from the API.
+	recvOffset, err := result.GetResult(ctx)
```

Comment:
Maybe best to add all the `GetResult` calls to the end to show off that multiple appends can get queued at a time?

Reply:
Switched to a slice of appends which get checked after the appends are all sent.